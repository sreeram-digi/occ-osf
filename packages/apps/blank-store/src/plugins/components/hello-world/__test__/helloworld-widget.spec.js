/*
 ** Copyright (c) 2020 Oracle and/or its affiliates.
 */

import React from 'react';
import {act, render, waitFor, screen} from '@oracle-cx-commerce/test/component/custom-render';
import {createMockStore} from '@oracle-cx-commerce/test/component/create-mock-store';
import Widget from '@oracle-cx-commerce/react-components/widget';
import {preloadComponents} from '@oracle-cx-commerce/commerce-utils/react';
import {_HelloWorld} from '../../index';

test('<HelloWorld/> unit/widget level testing with @testing-library/react', async () => {
  // create a store with the state structure widget rendering needs.
  const mockStore = createMockStore({
    pageRepository: {
      widgets: {
        helloWorldWidget123: {
          helloWorld: 'Hello World',
          id: 'helloWorldWidget123',
          componentId: '_HelloWorld',
          regions: []
        }
      }
    }
  });
  // list of comps to preload (resolves dynamic imports of comps) in node env.
  const comps = {_HelloWorld};
  await preloadComponents(comps);
  // Render helloWorld Widget
  const {container} = await act(() => render(<Widget widgetId="helloWorldWidget123" />, mockStore, comps));
  // Expect screen to render Below text
  await waitFor(() => screen.getByText('Hello World'));
  // Take a snapshot of the widget and store it.
  expect(container).toMatchSnapshot();
});
