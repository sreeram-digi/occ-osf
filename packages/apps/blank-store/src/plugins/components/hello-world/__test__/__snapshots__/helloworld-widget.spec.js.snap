// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld/> unit/widget level testing with @testing-library/react 1`] = `
<div>
  <style
    data-component="HelloWorld"
  />
  <div
    class="HelloWorld"
  >
    Hello World
  </div>
</div>
`;
